### ISSUE TITLE
iOS 13 users don't get subscribed and onIds event does not get triggered

### ISSUE THREAD

**Issue by codecrusher84** [OneSignal/react-native-onesignal#887]
**Description:**

Users that updated their devices to iOS 13 and above do not get subscribed to onesignal and onIds event does not get triggered even though users allow push notifications on app start.
Push notifications are working fine on both android and iOS versions below 13, the users get subscribed to onesignal with playerId and are receiving push notifications.

Further more the simulator with iOS version 13 gets subscribed and receives the onIds event with playerid.

**Environment**

1. React Native v0.59.5
2. react-native-onesignal v3.3.2 (updated from v3.2.12  since this version was the first when the issue was noticed)
3. SDK installed using react-native link react-native-onesignal

**Steps to Reproduce Issue:**

1. Install the OneSignal SDK using npm into your project
2. Run the app on iOS 13 or above

**Anything else:**
At this point I do not wish to update the React native version to greater than v0.59.5

**Comment by ayeshasaeed94** [comment]
Use latest version of react-native-onesignal which is 3.4.2 till current date. 
"react-native": "0.59.10"
"react-native-onesignal": "^3.4.2"

**Comment by tholhubner** [comment]
I am now seeing this issue as well. We are on RN 0.59.10 and the one signal package is "^3.4.2" but none of our iOS 13 users are being registered. All iOS 12 and earlier are working as expected. Is there any advice on resolving this issue?

**Comment by kylethielk** [comment]
We are also seeing this exact same behavior.

We are on react-native-onesignal => 3.4.2. I've triple checked that OneSignal-IOS-SDK is at 2.11.2. We are seeing exactly the error from [OneSignal/OneSignal-iOS-SDK#504](https://github.com/OneSignal/OneSignal-iOS-SDK/issues/504) 

```
VERBOSE: network response (OSRequestRegisterUser): {
    errors =     (
        "[\"Identifier invalid format.\"]"
    );
    success = 0;
}
```

i've tried downgrading to 3.4.1, completely removed OneSignal and re-added but cannot get this working on iOS 13.

It works fine on Android + iOS 12. 

Anyone with any luck?

**Comment by kylethielk** [comment]
> We are also seeing this exact same behavior.
> 
> We are on react-native-onesignal => 3.4.2. I've triple checked that OneSignal-IOS-SDK is at 2.11.2. We are seeing exactly the error from [OneSignal/OneSignal-iOS-SDK#504](https://github.com/OneSignal/OneSignal-iOS-SDK/issues/504)
> 
> ```
> VERBOSE: network response (OSRequestRegisterUser): {
>     errors =     (
>         "[\"Identifier invalid format.\"]"
>     );
>     success = 0;
> }
> ```
> 
> i've tried downgrading to 3.4.1, completely removed OneSignal and re-added but cannot get this working on iOS 13.
> 
> It works fine on Android + iOS 12.
> 
> Anyone with any luck?

For anyone stumbling on this, I highly recommend clearing CocoaPods cache, xCode Derived data and any other caches you might have. After fully analyzing the OneSignal logs I realized our build was picking up an old version of the iOS SDK. After clearing all caches, and confirming the correct library was being built iOS 13 began working.

**Comment by rgomezp** [comment]
Thanks for providing the solution @kylethielk 
I did notice most people complaining about this are on pre-0.60 versions of React-Native. If you are running Cocoapods on any of these versions of RN, auto-linking is not done automatically and so you have to make sure any changes to the modules in your node_modules directory are propagated to Cocoapods. It may be enough to run `react-native link react-native-onesignal` or perhaps going into your `ios` directory and running `pod update`.

Related: [how to manualy update the iOS OneSignal Native SDK in React Native](https://documentation.onesignal.com/docs/react-native-sdk-setup#section-manually-updating-ios-onesignalnativesdk)

**Comment by codecrusher84** [comment]
The solution for me was to update to react-native-onesignal v3.5.0
I did not use pod to install, only react-native-link which caused ios build to break at first because the libraries were not linked correctly.

I had to manually add RCTOneSignal.xcodeproj to libraries in xcode then link libRCTOneSignal.a to OneSignalNotificationServiceExtension target.
After that OneSignalNotificationServiceExtension target missed quite a bit of frameworks...
JavascriptCore, webkit, core graphics, core text, once I linked all of these ios build was successful and the notifications worked on all devices.

<img width="1406" alt="Screen Shot 2019-10-29 at 11 49 52 PM" src="https://user-images.githubusercontent.com/56540325/67815701-f6e20200-faa7-11e9-8480-ee3f2eb71164.png">
